cmd/cadvisor.go

```diff
@@ -153,6 +160,53 @@ func main() {
 		klog.Fatalf("Failed to register HTTP handlers: %v", err)
 	}
 
+	// Load TLS server configuration
+	var tlsConfig tls.Config
+	if *tlsCertPath != "" && *tlsKeyPath != "" {
+		if *tlsCertPath == "" || *tlsKeyPath == "" {
+			klog.Fatal("Both tls_cert_path and tls_key_path are required at the same time")
+		}
+
+		// Verify that the key/certificate load and are valid before starting up
+		_, err := tls.LoadX509KeyPair(*tlsCertPath, *tlsKeyPath)
+		if err != nil {
+			klog.Fatalf("Failed to load TLS certificate/key pair: %v", err)
+		}
+
+		tlsConfig.GetCertificate = func(*tls.ClientHelloInfo) (*tls.Certificate, error) {
+			cert, err := tls.LoadX509KeyPair(*tlsCertPath, *tlsKeyPath)
+			if err != nil {
+				return nil, err
+			}
+			return &cert, nil
+		}
+	}
+
+	if *tlsCAPath != "" {
+		clientCAPool := x509.NewCertPool()
+		clientCAFile, err := os.ReadFile(*tlsCAPath)
+		if err != nil {
+			klog.Fatalf("Failed to load TLS CA file: %v", err)
+		}
+		clientCAPool.AppendCertsFromPEM(clientCAFile)
+		tlsConfig.ClientCAs = clientCAPool
+	}
+
+	switch *tlsClientAuth {
+	case "request":
+		tlsConfig.ClientAuth = tls.RequestClientCert
+	case "optional":
+		tlsConfig.ClientAuth = tls.VerifyClientCertIfGiven
+	case "requireany":
+		tlsConfig.ClientAuth = tls.RequireAnyClientCert
+	case "require":
+		tlsConfig.ClientAuth = tls.RequireAndVerifyClientCert
+	case "none":
+		tlsConfig.ClientAuth = tls.NoClientCert
+	default:
+		klog.Fatalf("Invalid tls_client_auth: %s", *tlsClientAuth)
+	}
+
```

Comment:
Extracting this whole part to a separate function would make code a bit more readable. You would just need to return `*tls.Config`.

Reply:
Done